bigtable/internal/testproxy/proxy.go

```diff
@@ -0,0 +1,626 @@
+		cq := bigtable.ColumnFilter(string(re))
+		f = &cq
+
+	case *btpb.RowFilter_ColumnRangeFilter:
+		crf := fpb.(*btpb.RowFilter_ColumnRangeFilter)
+		cr := crf.ColumnRangeFilter
+
+		var start, end string
+		switch sf := cr.StartQualifier; sf.(type) {
+		case *btpb.ColumnRange_StartQualifierOpen:
+			start = string(sf.(*btpb.ColumnRange_StartQualifierOpen).StartQualifierOpen)
+		case *btpb.ColumnRange_StartQualifierClosed:
+			start = string(sf.(*btpb.ColumnRange_StartQualifierClosed).StartQualifierClosed)
+		}
+
+		switch ef := cr.EndQualifier; ef.(type) {
+		case *btpb.ColumnRange_EndQualifierClosed:
+			end = string(ef.(*btpb.ColumnRange_EndQualifierClosed).EndQualifierClosed)
+		case *btpb.ColumnRange_EndQualifierOpen:
+			end = string(ef.(*btpb.ColumnRange_EndQualifierOpen).EndQualifierOpen)
+		}
+
+		cf := bigtable.ColumnRangeFilter(cr.FamilyName, start, end)
+		f = &cf
+
+	case *btpb.RowFilter_TimestampRangeFilter:
+		trf := fpb.(*btpb.RowFilter_TimestampRangeFilter)
+		tsr := trf.TimestampRangeFilter
+
+		tsf := bigtable.TimestampRangeFilter(time.UnixMicro(tsr.StartTimestampMicros), time.UnixMicro(tsr.EndTimestampMicros))
+		f = &tsf
+
+	case *btpb.RowFilter_ValueRegexFilter:
+		vrf := fpb.(*btpb.RowFilter_ValueRegexFilter)
+		re := vrf.ValueRegexFilter
+		vr := bigtable.ValueFilter(string(re))
+		f = &vr
+
+	case *btpb.RowFilter_ValueRangeFilter:
+		vrf := fpb.(*btpb.RowFilter_ValueRangeFilter)
+
+		var start, end []byte
+		switch sv := vrf.ValueRangeFilter.StartValue; sv.(type) {
+		case *btpb.ValueRange_StartValueOpen:
+			start = sv.(*btpb.ValueRange_StartValueOpen).StartValueOpen
+		case *btpb.ValueRange_StartValueClosed:
+			start = sv.(*btpb.ValueRange_StartValueClosed).StartValueClosed
+		}
+
+		switch ev := vrf.ValueRangeFilter.EndValue; ev.(type) {
+		case *btpb.ValueRange_EndValueOpen:
+			end = ev.(*btpb.ValueRange_EndValueOpen).EndValueOpen
+		case *btpb.ValueRange_EndValueClosed:
+			end = ev.(*btpb.ValueRange_EndValueClosed).EndValueClosed
+		}
+
+		vr := bigtable.ValueRangeFilter(start, end)
+		f = &vr
+
+	case *btpb.RowFilter_CellsPerRowOffsetFilter:
+		cof := fpb.(*btpb.RowFilter_CellsPerRowOffsetFilter)
+		off := cof.CellsPerRowOffsetFilter
+		co := bigtable.CellsPerRowOffsetFilter(int(off))
+		f = &co
+
+	case *btpb.RowFilter_CellsPerRowLimitFilter:
+		clf := fpb.(*btpb.RowFilter_CellsPerRowLimitFilter)
+		lim := clf.CellsPerRowLimitFilter
+		cl := bigtable.CellsPerRowLimitFilter(int(lim))
+		f = &cl
+
+	case *btpb.RowFilter_CellsPerColumnLimitFilter:
+		ccf := fpb.(*btpb.RowFilter_CellsPerColumnLimitFilter)
+		lim := ccf.CellsPerColumnLimitFilter
+		cc := bigtable.LatestNFilter(int(lim))
+		f = &cc
+
+	case *btpb.RowFilter_StripValueTransformer:
+		sv := bigtable.StripValueFilter()
+		f = &sv
+
+	case *btpb.RowFilter_ApplyLabelTransformer:
+		alf := fpb.(*btpb.RowFilter_ApplyLabelTransformer)
+		l := alf.ApplyLabelTransformer
+		al := bigtable.LabelFilter(l)
+		f = &al
+	default:
```

Comment:
Since you're returning `f` below which would be `nil` if any of these case statements don't match, you can remove this branch OR just put a return in each case statement and remove the `return f` below. I personally find returning out of case statements easier to read when it gets this long.

Reply:
The compiler doesn't like the missing return statement, so I just deleted the `default` branch.